crates/rethnet_eth/src/remote/client.rs

```diff
@@ -372,66 +287,441 @@ impl RpcClient {
                 )
             })?;
 
-        let code =
-            serde_json::from_value::<jsonrpc::Response<jsonrpc::ZeroXPrefixedBytes>>(code_response)
-                .map_err(|err| RpcClientError::DeserializationError {
-                    msg: err.to_string(),
-                    body: response.text.clone(),
-                })
-                .and_then(|response| {
-                    response.data.into_result().map_or_else(
-                        |error| {
-                            Err(RpcClientError::JsonRpcError {
-                                error,
-                                request: code_request,
-                            })
-                        },
-                        |bytes| Ok(Bytecode::new_raw(bytes.inner)),
-                    )
-                })?;
+        let code =
+            serde_json::from_value::<jsonrpc::Response<jsonrpc::ZeroXPrefixedBytes>>(code_response)
+                .map_err(|err| RpcClientError::DeserializationError {
+                    msg: err.to_string(),
+                    body: response.text.clone(),
+                })
+                .and_then(|response| {
+                    response.data.into_result().map_or_else(
+                        |error| {
+                            Err(RpcClientError::JsonRpcError {
+                                error,
+                                request: code_request,
+                            })
+                        },
+                        |bytes| {
+                            Ok(if bytes.inner.is_empty() {
+                                None
+                            } else {
+                                Some(Bytecode::new_raw(bytes.inner))
+                            })
+                        },
+                    )
+                })?;
+
+        Ok(AccountInfo {
+            balance,
+            code_hash: code.as_ref().map_or(KECCAK_EMPTY, Bytecode::hash),
+            code,
+            nonce,
+        })
+    }
+
+    /// eth_getBlockByHash
+    pub async fn get_block_by_hash(
+        &self,
+        hash: &B256,
+        include_transactions: bool,
```

Comment:
passing `false` for `include_transactions` causes a `panic`

Reply:
Good catch. I added the proper parsing code for dealing with transactions being hashes